cvat-ui/src/components/annotation-page/standard-workspace/objects-side-bar/labels-list.tsx

```diff
@@ -1,26 +1,110 @@
-// Copyright (C) 2020 Intel Corporation
+// Copyright (C) 2020-2021 Intel Corporation
 //
 // SPDX-License-Identifier: MIT
 
-import React from 'react';
+import React, { useCallback, useState } from 'react';
+import { useDispatch, useSelector } from 'react-redux';
+import message from 'antd/lib/message';
 
+import { CombinedState } from 'reducers/interfaces';
+import { rememberObject, updateAnnotationsAsync } from 'actions/annotation-actions';
 import LabelItemContainer from 'containers/annotation-page/standard-workspace/objects-side-bar/label-item';
+import GlobalHotKeys from 'utils/mousetrap-react';
 
-interface Props {
-    labelIDs: number[];
-    listHeight: number;
-}
+function LabelsListComponent(): JSX.Element {
+    const dispatch = useDispatch();
+    const {
+        annotation: {
+            job: { labels },
+            tabContentHeight: listHeight,
+            annotations: { activatedStateID, states },
+        },
+        shortcuts: { keyMap },
+    } = useSelector((state: CombinedState) => state);
+    const labelIDs = labels.map((label: any): number => label.id);
+
+    const [keyToLabelMapping, setKeyToLabelMapping] = useState<Record<string, number>>(
+        Object.fromEntries(
+            labelIDs.slice(0, 10).map((labelID: number, idx: number) => [idx + 1 > 9 ? 0 : idx + 1, labelID]),
```

Comment:
I guess it will like nicer `(idx + 1) % 10`

Reply:
Yes, nice idea